clang/lib/AST/ParentMapContext.cpp

```diff
@@ -427,42 +423,53 @@ class ParentMapContext::ParentMap::ASTVisitor
     return Result;
   }
 
-  bool TraverseDecl(Decl *DeclNode) {
+  bool TraverseDecl(Decl *DeclNode) override {
     return TraverseNode(
-        DeclNode, DeclNode, [&] { return VisitorBase::TraverseDecl(DeclNode); },
+        DeclNode, DeclNode,
+        [&] { return DynamicRecursiveASTVisitor::TraverseDecl(DeclNode); },
         &Map.PointerParents);
   }
-  bool TraverseTypeLoc(TypeLoc TypeLocNode) {
+  bool TraverseTypeLoc(TypeLoc TypeLocNode) override {
     return TraverseNode(
         TypeLocNode, DynTypedNode::create(TypeLocNode),
-        [&] { return VisitorBase::TraverseTypeLoc(TypeLocNode); },
+        [&] {
+          return DynamicRecursiveASTVisitor::TraverseTypeLoc(TypeLocNode);
```

Comment:
Does this make sense? It's not the same, isn't it? The previous code would call back into `ParentMapContext`, but the new one doesn't(?) This should basically be `this->TraverseTypeLoc()`, right? Am I missing something?

Reply:
Er, unless I misread something horribly somewhere, in both cases this is statically calling the base class implementation (`VisitorBase` was `RecursiveASTVisitor<ASTVisitor>`, i.e. the base class of this thing).